### ISSUE TITLE
[clang-tidy] `llvmlibc-inline-function-decl` erroneously reports on nested templates

### ISSUE THREAD

**Issue by jhuber6** [llvm/llvm-project#62746]
The `llvmlibc-inline-function-decl` check currently reports erroneously  on nested template parameters. Here is a reproducer.
```c++
#define LIBC_INLINE inline

namespace __llvm_libc {

template <typename T> struct S {
  template <typename U> void foo();
};

template <typename T> template <typename U> LIBC_INLINE void S<T>::foo() {}
} // namespace __llvm_libc
```

```console
$ clang-tidy test.h --checks=-'*,llvmlibc*'
2 warnings generated.
/home/jhuber/Documents/llvm/llvm-project/build/test.h:9:1: warning: 'foo' must be tagged with the LIBC_INLINE macro; the macro should be placed at the beginning of the declaration [llvmlibc-inline-function-decl]
template <typename T> template <typename U> LIBC_INLINE void S<T>::foo() {}
^
```

**Comment by AMS21** [comment]
https://reviews.llvm.org/D153218
